File: src/EChat.Frontend/src/app/modules/chat/dtos/message.ts
import { ChatMessage } from "..";

export interface Message {
    name: string;
    text: string;
    userId: string;
    sentiment: string;
}

export function createDefaultMessage(): Message {
    return {
        text: "",
        name: "",
        userId: "",
        sentiment: ""
    };
}

export function mapMessageToChatMessage(message: Message, userId: string): ChatMessage {
    const isSent = message.userId === userId;

    return {
        text: message.text,
        userName: message.name,
        isSent: isSent,
        sentiment: message.sentiment
    };
}

File: src/EChat.Backend/ChatApi/Dtos/Message.cs
﻿namespace ChatApi.Dtos
{
    public class Message
    {
        public string Name { get; set; } = string.Empty;
        public string Text { get; set; } = string.Empty;
        public string UserId { get; set; } = string.Empty;
        public string Sentiment { get; set; } = "Unknown";
    }
}
